jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/BaseEnricherTest.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.enricher.api;
+
+import org.eclipse.jkube.kit.config.resource.ResourceConfig;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Answers.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BaseEnricherTest {
+  private BaseEnricher baseEnricher;
+  private ResourceConfig resourceConfig;
+  private JKubeEnricherContext context;
+
+  @Before
+  public void setup() {
+    resourceConfig = mock(ResourceConfig.class, RETURNS_DEEP_STUBS);
+    context = mock(JKubeEnricherContext.class, RETURNS_DEEP_STUBS);
+    baseEnricher = new BaseEnricher(context, "base-enricher");
+  }
+
+  @Test
+  public void getImagePullPolicy_whenNoConfigPresent_shouldReturnNull() {
+    // Given + When
+    String value = baseEnricher.getImagePullPolicy(null, null);
+
+    // Then
+    assertThat(value).isNull();
```

Comment:
Shouldn't a default value be returned always?

Reply:
default value is provided by Enricher's Config and is passed into method's second argument. I can update second argument to be non-null value and verify that it's returned.

Reply:
If the default value is common to both Enrichers, then it might be better to provide it from the BaseEnricher (for consistency with the getReplicas method) ---> remove the second parameter